apps/dashboard/lib/trpc/routers/key/create.ts

```diff
@@ -100,7 +103,22 @@ export const createKey = t.procedure
             "We are unable to create the key. Please contact support using support.unkey.dev",
         });
       });
-
+    if (input.recoverEnabled && keyAuth?.storeEncryptedKeys) {
```

Comment:
This doesn't defensively handle if the recoverEnabled is true but the storeEncryptedKeys is false (99% of cases) so the user doesn't receive and error and believes they would have successfully created a recoverable key.

Reply:
If recoverEnabled is true and storeEncryptedKeys is false following error will be displayed

![image](https://github.com/user-attachments/assets/54622bd0-7a03-4727-b5ca-bd0856f313de)